doc/Coding Guidelines.md

```diff
@@ -0,0 +1,107 @@
+# Vue Storefront Coding Guidelines - DRAFT
+
+Hello Extension / Theme / Core developer! This is just a short guide to align the code style among the core modules and all the extensions and thems.
+
+## General guidelines
+In general we're using kind of mix of: [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) and [JavaScript Standard Style](https://github.com/feross/standard/blob/master/RULES.md#javascript-standard-style). There are a eslint and vue-eslint configured to enforce the style. Please do use the same [configs in Your modules](https://github.com/DivanteLtd/vue-storefront/blob/master/.eslintrc.js). 
+
+## ES2015
+We do preffer to use ES2015 constructions in the code - and there is a babel compiler set to allow us to do so. So use the ES modules instead of `require` and feel free to use other fancy structures.
+
+## Components
+Components should be created as [Single File Vue components](https://vuejs.org/v2/guide/single-file-components.html) and names using `CamelCase.vue`.
+All the component properties should be named using `camelCase` as well.
+All the component data fields should be `camelCased`.
+
+Theme catalog structure is well described in here: [Working with themes](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/themes/Working%20with%20themes.md)
+
+## Extensions
+For the reference of how extension can be structured please take a look at [template](https://github.com/DivanteLtd/vue-storefront/tree/develop/src/extensions/template).
+Check the [Working with extensions](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/extensions/Working%20with%20extensions.md) for more info.
+
+It's something like this:
+
+```
+|____store.js
+|____index.js
+|____components
+| |____ComponentExample.vue
+|____router.js
+|____package.json
+|____pages
+| |____PageExample.vue
+```
+
+## Events
+We're using [Vue events](https://vuejs.org/v2/guide/events.html) for async. operations handling and exposing some extension points in the code.
```

Comment:
These part of Vue docs are related to handling typical DOM events like `click` or `change`.

In our case, we should link to https://vuejs.org/v2/guide/components.html#Using-v-on-with-Custom-Events and https://vuejs.org/v2/api/#Instance-Methods-Events + maybe also some docs about creating EventBus. 

But... I'd suggest to get rid of this concept and move everything to Vuex, b/c using two ways of managing app state is not necessary and not easy to understand / work with.

Reply:
Agree. We should use Vuex not only b/c its easier to maintain but its also easier to understand. Also currently we are depending too much on the events. I tried to change some things yesterday with @qiqqq  ago and it's worse than visiting hell now b/c of multilevel event handling. It would be much cleaner if we will group logic in smart components and keep the app state in Vuex. Some of the events should stay but most of them is redundant